doc/specs/#980 - SnapOnOutput.md

```diff
@@ -0,0 +1,100 @@
+- `atBottom`: new output causes the viewport to update **IF** the viewport is already at the virtual bottom
+- `always`: new output causes the viewport to update to the bottom of the scroll region
+
+The `TerminalCore` is responsible for moving the viewport on a scroll event. All of the logic for this feature should be handled here.
+
+A new private enum array `_snapOnOutput` will be introduced to save which of these settings are included. The `_NotifyScrollEvent()` calls (and nearby code) will be surrounded by conditional checks for the enums above. This allows it to be used to determine if the viewport should update given a specific situation.
+
+## UI/UX Design
+
+The `snapOnOutput` setting is done at a profile-level to be near `snapOnInput`. Additionally, the `never` value seems more valuable when the user can dedicate a specific task to the profile. Such a scenario would be a shell that frequently generates new output (i.e.: a live-generating log), but the user is not necessarily interested in what the latest output is.
+
+The default `snapOnOutput` value will be `[ "noSelection", "atBottom" ]`.
+
+When an enum array is defined in the settings, it will be interpreted logically. The following scenarios will be interpreted as follows:
+- `[ "always", "atBottom" ]` --> `"always"` because `atBottom` is a subset of `always`
+- `[ "never", "atBottom" ]` --> `"atBottom"`
+
+## Capabilities
+
+### Accessibility
+
+The `UiaRenderer` exposes new content that is rendered to the screen upon a scroll. No additional changes are necessary here.
+
+### Security
+
+N/A
+
+### Reliability
+
+N/A
+
+### Compatibility
+
+N/A
+
+### Performance, Power, and Efficiency
+
+N/A
+
+## Potential Issues
+
+### Circling the buffer
+If the text buffer fills up, the text buffer begins 'circling'. This means that new output shifts lines of the buffer up to make space. In a case like this, if `snapOnOutput` is set to `never`, the viewport should actually scroll up to keep the same content on the viewport.
+
+In the event that the buffer is circling and the viewport has been moved to the top of the buffer, that content of the buffer is now lost (as the 'Infinite Scrollback' feature does not exist or is disabled). At that point, the viewport will remain at the top of the buffer and the new output will push old output out of the buffer.
+
+### Infinite Scrollback
+See **Future considerations** > **Infinite Scrollback**.
+
+## Future considerations
+
+### Extensibility
+The introduction of `enum SnapOnOutput` allows for this feature to be enabled/disabled in more complex scenarios. A potential extension would be to introduce a new UI element or keybinding to toggle this feature.
+
+Another would be to introduce additional accepted values. Consider the output from a large testing suite. A user could create an extension that snaps on output for failing tests or when the testing has concluded.

```

Comment:
Introducing new enum values isn't the way to do this. I might suggest dropping this part of this section.